docs/specs/servers/relay/relay-webhooks.md

```diff
@@ -0,0 +1,26 @@
+
+The webhook registration is remembered by the relay for the `exp` of the registration JWT with a maximum of 30 days, and must be renewed periodically or else the registration will expire and watch events will no longer be delivered. If a watched client registers a webhook with the same service URL and webhook URL, then it will replace the existing webhook. This can be used to renew the webhook registration, or change the webhook parameters. A watched client can have a maximum of 64 webhooks registered at once. Attempting to register more than 64 webhooks will return an error.
+
+The diagram below shows the flow of a message from a publisher to subscriber, and the 4 potential watch events it could trigger.
+
+![Webhooks message flow](/assets/webhooks.png)
+
+`client_id` is the client ID of the watched client. Watch events are triggered when this client ID either publishes or is subscribed to receive a message. Webhook events cannot be filtered by topic, and can only be filtered by status, type, and tag.
+
+Webhooks can be setup to trigger on one or more of these events. Available statuses are `accepted`, `queued`, and `delivered`. Available types are `publisher` and `subscriber`. Valid combinations of these statuses and types are:
+
+- `accepted` & `publisher`: The client publishes a message, regardless if any subscribers are subscribed to the topic or will receive the message.
+- `accepted` & `subscriber`: The client has received the message or the message has been stored in the mailbox.
+- `queued` & `subscriber`: The client is subscribed to the message, but because it does not have a websocket connection open to receive the message immediately, the message has been stored in the mailbox instead.
+- `delivered` & `subscriber`: The client received the message by being connected via websocket when the message was published, connected to the websocket and received the message from the mailbox, or called `irn_fetchMessages` or `irn_batchFetchMessages`.
+
+If a watched client never receives the message, the message will persist in the mailbox for the TTL of the message. So if webhooks are exclusively used to deliver messages to the watched client, rather than another method (such as websocket or `irn_fetchMessages`) then it is considered a best practice to pair this with the usage of the [`irn_batchReceive`](./relay-server-rpc.md#batch-receive) RPC method to clear the mailbox of all messages so it doesn't consume resources on the relay unnecessarily.
```

Comment:
```suggestion
If a watched client never receives the message, the message will persist in the mailbox for the TTL of the message (as it would normally). So if webhooks are exclusively used to deliver messages to the watched client, rather than another method (such as websocket or `irn_fetchMessages`) then it is considered a best practice to pair this with the usage of the [`irn_batchReceive`](./relay-server-rpc.md#batch-receive) RPC method to clear the mailbox of all messages so it doesn't consume resources on the relay unnecessarily.
```